core/schema/module.graphqls

```diff
@@ -114,18 +110,33 @@ extend type Directory {
     directory.
     """
     sourceSubpath: String
-
-    """
-    A pre-built runtime container to use instead of building one from the
-    source code. This is useful for bootstrapping.
-
-    You should ignore this unless you're building a Dagger SDK.
-    """
-    runtime: ContainerID
   ): Module!
 }
 
-input ModuleEnvironmentVariable {
+"Static configuration for a module (e.g. parsed contents of dagger.json)"
+type ModuleConfig {
+  "The name of the module."
   name: String!
-  value: String
+
+  "The root directory of the module's project, which may be above the module source code."
+  root: String
+
+  "Either the name of a built-in SDK ('go', 'python', etc.) OR a module reference pointing to the SDK's module implementation."
+  sdk: String!
+
+  "Include only these file globs when loading the module root."
+  include: [String!]
+
+  "Exclude these file globs when loading the module root."
+  exclude: [String!]
+
+  "Modules that this module depends on."
+  dependencies: [String!]
+}
+
+extend type Query {
+  """
+  Load the static configuration for a module from the given source directory and optional subpath.
+  """
+  moduleConfig(sourceDirectory: DirectoryID!, subpath: String): ModuleConfig!
```

Comment:
Only in an integ test so far. I added this because Helder said he'll want to know the name of the module for follow-up improvements to the Python SDK module implementation. I just wanted to handle that right away to make sure it was gonna work with the new code in this PR and thus avoid another refactor or something.

For a little more context, these are the other options I could think of besides what I went with:
1. Just add `moduleName` as a parameter to the `ModuleRuntime`+`Codegen` functions
   * Felt like the beginning of an eternal game of whack-a-mole
2. Just give a single parameter `*Module` as input
   * Cool idea in that you now have a Module that accepts another Module as input, updates it and returns the modified one.
   * But less cool in that the module the SDKs accept would be in an ill-defined state of initialization, hard to understand
3. Add `ModuleConfig` to the API.
   * Could have also provided `cfg *ModuleConfig` as an input arg to the `ModuleRuntime`+`Codegen` functions, but went with this approach of putting it as an API call partially to avoid the current questions around handling of input types in the Go SDK and partially because this API seemed more potentially useful in the long run

Either way, totally agree that we should probably integrate this better into the rest of the API. `Module.config` makes sense too. Better for a follow up though